mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb

```diff
@@ -3814,6 +3876,114 @@ resource "google_container_cluster" "with_node_config" {
 `, clusterName)
 }
 
+func testAccContainerCluster_withNodeConfigReservationAffinity(clusterName string) string {
+	return fmt.Sprintf(`
+resource "google_container_cluster" "with_node_config" {
+  name               = "%s"
+  location           = "us-central1-f"
+  initial_node_count = 1
+
+  node_config {
+    machine_type    = "e2-medium"
+    disk_size_gb    = 15
+    disk_type       = "pd-ssd"
+    oauth_scopes = [
+      "https://www.googleapis.com/auth/monitoring",
+      "https://www.googleapis.com/auth/compute",
+      "https://www.googleapis.com/auth/devstorage.read_only",
+      "https://www.googleapis.com/auth/logging.write",
+    ]
+    service_account = "default"
+    metadata = {
+      foo                      = "bar"
+      disable-legacy-endpoints = "true"
+    }
+    labels = {
+      foo = "bar"
+    }
+    tags             = ["foo", "bar"]
+    preemptible      = true
+
+    // Updatable fields
+    image_type = "COS_CONTAINERD"
+
+    reservation_affinity {
+      consume_reservation_type = "ANY_RESERVATION"
+    }
+  }
+}
+`, clusterName)
+}
+
+func testAccContainerCluster_withNodeConfigReservationAffinitySpecific(clusterName string) string {
+	return fmt.Sprintf(`
+
+resource "google_project_service" "compute" {
+  service = "compute.googleapis.com"
+}
+
+resource "google_project_service" "container" {
+  service = "container.googleapis.com"
+  depends_on = [google_project_service.compute]
```

Comment:
The current failure is:

```
The service container.googleapis.com is depended on by the following active service(s): binaryauthorization.googleapis.com,composer.googleapis.com,datafusion.googleapis.com,file.googleapis.com,memcache.googleapis.com,redis.googleapis.com,tpu.googleapis.com; Please specify disable_dependent_services=true if you want to proceed with disabling all services.
```

This is because the service is being enabled on the default test project, which has a bunch of other apis enabled. If you want to run on the default test project you need to disable disable_on_destroy:

```suggestion
  depends_on = [google_project_service.compute]
  disable_on_destroy = false
```

[Another test that does it this way](https://github.com/GoogleCloudPlatform/magic-modules/blob/36335236c589b09544e67d6e84f7610e3ab7fcc6/mmv1/third_party/terraform/tests/resource_gke_hub_feature_test.go.erb#L272)

The other option would be to provision a new project just for this test. [Here's a test that takes that approach](https://github.com/GoogleCloudPlatform/magic-modules/blob/36335236c589b09544e67d6e84f7610e3ab7fcc6/mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb#L4982).

I think the first option is simpler and should be fine. The second case is more complicated since it involves cross-project interactions.

Reply:
Added this to both compute and container apis